File: typescript/hamming/hamming.ts
export default class Hamming {
  compute(a: string, b: string): number {
    if (a.length != b.length) {
      throw 'DNA strands must be of equal length.';
    }

    let result = 0;

    for (let i = 0; i < a.length; i++) {
      if (a[i] != b[i]) {
        result++;
      }
    }

    return result;
  }
}

File: dart/hamming/lib/hamming.dart
class Hamming {
  // Put your code here

  int distance(String left, String right) {
    if ((left.length == 0 || right.length == 0) &&
        left.length + right.length != 0) {
      throw ArgumentError('no strand must be empty');
    }

    if (left.length != right.length) {
      throw ArgumentError('left and right strands must be of equal length');
    }

    int res = 0;

    for (var i = 0; i < left.length; i++) {
      if (left[i] != right[i]) {
        res++;
      }
    }

    return res;
  }
}
